packages/core/src/loaders/enhancedVolumeModifiers/applyEnhancedVolumeModifiers.ts

```diff
@@ -0,0 +1,19 @@
+import type { ImageVolumeProps } from '../../types';
+import type {
+  EnhancedVolumeModifier,
+  EnhancedVolumeModifierContext,
+} from './types';
+
+/**
+ * Runs every modifier sequentially and returns the final props.
+ */
+export function applyEnhancedVolumeModifiers(
+  baseProps: ImageVolumeProps,
+  modifiers: EnhancedVolumeModifier[],
+  context: EnhancedVolumeModifierContext
+): ImageVolumeProps {
+  return modifiers.reduce(
+    (currentProps, modifier) => modifier.apply(currentProps, context),
```

Comment:
i don't understand all these complexity, do you have a bigger plan? since all these tiny files can be helpr functions within the same volume loader